edwards/ed25519/eddsa.go

```diff
@@ -0,0 +1,107 @@
+package ed25519
+
+import (
+	"crypto/sha512"
+	"errors"
+	"fmt"
+
+	"github.com/dedis/crypto/abstract"
+)
+
+// EdDSASign will return a EdDSA signature using Ed25519. The secret must be
+// Picked() so it can use the prefix, the right part of the hashing done in
+// Pick.
+// Takes
+// - suite is the suite to use
+// - a secret to use to sign the message
+// - A corresponding public key
+// - msg message to sign
+// NOTE: Code taken from the Python implementation from the RFC
+// https://tools.ietf.org/html/draft-josefsson-eddsa-ed25519-02
+func EdDSASign(suite abstract.Suite, a abstract.Secret, A abstract.Point, msg []byte) ([]byte, error) {
```

Comment:
Should we take a generic suite? Or does this only work with the eddsa-suite?
What about taking the `suite.Hash` for hashing?